sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

```diff
@@ -24,92 +29,131 @@ enum class architecture {
   //   - the unique ID of the new architecture in SYCL RT source code to support
   //     querying the device architecture
   //
-  x86_64,
-  intel_cpu_spr,
-  intel_cpu_gnr,
-  intel_gpu_bdw,
-  intel_gpu_skl,
-  intel_gpu_kbl,
-  intel_gpu_cfl,
-  intel_gpu_apl,
+  // Important note about keeping architecture IDs below unique:
+  //   - the architecture ID must be a hex number with 16 digits
+  //   - the architecture ID must suit the following template:
+  //     0x AAAA BB CCCCCCCC DD (without spaces), where
+  //       - AAAA is 4-digit PCI vendor ID of that architecture
+  //       - BB is 2-digit unique number of vendor's accelerator. It is 00 by
+  //         default. E.g., for Intel, the PCI vendor ID is the same for all
+  //         accelerators: for CPU and for GPU. In this case BB can be equal to
+  //         00 for CPU and 01 for GPU
+  //       - AAAABB number must be unique to the only one category from
+  //         arch_category enum below. Two or more categories cannot share the
+  //         same AAAABB number
+  //       - CCCCCCCC is 8-digit number of architecture itself. It must be
+  //         unique for all architectures inside the category
+  //       - DD is 2-digit number reserved for future unexpected modifications
+  //         to keep uniqueness. It should be always 00 for now
```

Comment:
What's the advantage of using the PCI vendor ID here?  Couldn't we just use the top 32 bits to uniquely identify each device "family" with our own values?  For example, "Intel CPU family" could be 0, "Intel GPU family" could be 1, etc.  That would give us more flexibility and would reserve more bits for possible future use.

Reply:
Actually. no advantage. Yeah, changed to 1, 2, etc, but reserved 2 digits in case more than 16 architectures will be here (which is unlikely, but still)
BTW, should we include arch::x86_64 to intel_cpu category? Or it will be completely separate as it can be AMD CPU as well?

Reply:
I'm not sure what to do with `arch::x86_64` yet.  This is probably something that will change before we move this extension out of "experimental" status.  Let's leave it the way it is for now.